### PR TITLE
Contact Form: disable buttons when the form is submitted

### LINKED ISSUE #18708: Contact Form: submit button can be pressed multiple times, resulting in multiple submissions
#### Steps to reproduce the issue

1. Create Wordpress installation
2. Select twentyseventeen theme
3. Enable Jetpack
4. Make contact form
5. Test contact form. Press submit button as many times as possible before the page changes
6. Receive multiple copies of the form submission via email

#### What I expected

I expected the button to become disabled upon successful submission (no form errors).

AFAIK, this functionality was working in previous versions of wordpress/jetpack/twentyseventeen.

#### What happened instead
The button is not disabled, and can be clicked as many times as the user can accomplish.

For example, if unable to reproduce see URL: addiform DOT com
Try not to spam it too much.

### PR DESCRIPTION
**Pull request by kbrown9**
Fixes #18708

#### Changes proposed in this Pull Request:
Currently, the contact form can be clicked multiple times, resulting in multiple form submissions. Fix this by disabling the submitted form's button elements when a form is submitted.

* Create a new method, `get_script_for_form`, which adds an embedded script to the contact form html.
* In the script, disable the form's button elements when an element with the `contact-form` class is submitted.

Alternatives Considered
* I considered creating a new external file for the script, but the script is small and focused so an embedded script seemed fine. (It appears the contact form doesn't already load an external JS file.)

#### Jetpack product discussion
* n/a

#### Does this pull request change what data or activity we track or use?
* no

#### Testing instructions:
1. Install, activate, and connect this branch of Jetpack.
2. In wp-admin, create a new post and add a Contact Form block to the post.
3. Publish the new post.
4. Navigate to the new post. Fill out the contact form and submit it. The contact form should be submitted successfully.
5. In wp-admin, navigate to the Feedback menu item and confirm that the contact form submission was received.
6. Repeat steps 4-5. This time, attempt to click the Submit button multiple times.
7. In wp-admin, navigate to the Feedback menu item and confirm that the contact form submission was received once.
8. Repeat steps 4-5. This time, attempt to submit the form multiple times by tabbing to the Submit button and clicking 'enter' multiple times.
9. In wp-admin, navigate to the Feedback menu item and confirm that the contact form submission was received once.
10. Repeat steps 4-9 in different browsers (Chrome, Firefox, Safari).
11. Navigate to the new post on a mobile device. Fill out the contact form and attempt to click the submit button multiple times. The contact form should be submitted successfully.
12. In wp-admin, navigate to the Feedback menu item and confirm that the contact form submission was received once.

### CODE REVIEW

File: projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php [hunk 1]
```diff
@@ -2312,6 +2314,28 @@ static function success_message( $feedback_id, $form ) {
...
+	/**
+	 * Returns a script that disables the contact form button after a form submission.
+	 *
+	 * @return string The script.
+	 */
+	private static function get_script_for_form() {
+		return "<script>
+			( function () {
+				const contact_forms = document.getElementsByClassName('contact-form');
+				const buttons = document.getElementsByTagName('button');
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php:2326]: You're fetching all the buttons in the page here, not just the ones that belong to the form. Is that intended?

File: projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php [hunk 2]
```diff
@@ -2312,6 +2314,28 @@ static function success_message( $feedback_id, $form ) {
...
+
+				for ( form of contact_forms ) {
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php:2328]:
```suggestion
				for ( const form of contact_forms ) {
```

File: projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php [hunk 3]
```diff
@@ -2312,6 +2314,28 @@ static function success_message( $feedback_id, $form ) {
...
+					form.onsubmit = function() {
+						for( button of buttons ) {
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php:2330]:
```suggestion
						for( const button of buttons ) {
```

File: projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php (hunk 2 above)
Comment (kbrown9) [reply projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php]: Fixed in 008d53c.

File: projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php (hunk 3 above)
Comment (kbrown9) [reply projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php]: Fixed in 008d53c.

**Review by anomiex — APPROVED** [review]
Seemed to work correctly on latest versions of Firefox, Chromium, Android Firefox, and Android Chrome. Didn't have any other browsers handy to test with.

None of the browsers seemed to have a visual change for the "disabled" state of the button. Android Chrome still visually reacted to the multiple touches, but only one submission was recorded so that may have just been a visual artifact.

File: projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php (hunk 1 above)
Comment (kbrown9) [reply projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php]: Just to close the loop on this comment, I decided to disable only the submitted form's button elements in commit 008d53c. This means that the visitor could, in theory, still do things like submit multiple forms in the same page reload if the page has multiple forms. I think that behavior is acceptable. The same form being submitted multiple times is not acceptable behavior, and this PR prevents that from happening.

**Comment by kbrown9** [comment]
r234718-wpcom

**Pull request merged**